File: python_scripts/gene.py
import exon

class gene(object):

  def __init__(self, chromosome, start, end, name):
    self.__chromosome=chromosome
    self.__start_pos=start
    self.__end_pos=end
    self.__name=name

  def name(self):
    return self.__name

  def chromosome(self):
    return self.__chromosome
  
  def start_pos(self):
    return self.__start_pos

  def end_pos(self):
    return self.__end_pos

  def contains(self, this_exon):
    """
    returns True if the exon (object) passed to this method is contained in teh current gene. else, false
    """
    if this_exon.chromosome()==self.__chromosome:
      if this_exon.start_pos()>=self.__start_pos and this_exon.end_pos()<=self.__end_pos:
        return True
    return False

  def __str__(self):
    return str(self.__name)+','+str(self.__chromosome)+','+str(self.__start_pos)+','+str(self.__end_pos)

File: java/src/pojos/Gene.java
package pojos;

public class Gene {

	private String chromosome, geneName;
	private long start_pos, end_pos;
	
	public Gene(String chr, long start, long end, String name){
		this.chromosome=chr;
		this.start_pos=start;
		this.end_pos=end;
		this.geneName=name;
	}

	public String getChromosome() {
		return chromosome;
	}

	public void setChromosome(String chromosome) {
		this.chromosome = chromosome;
	}

	public String getGeneName() {
		return geneName;
	}

	public void setGeneName(String geneName) {
		this.geneName = geneName;
	}

	public long getStart_pos() {
		return start_pos;
	}

	public void setStart_pos(long start_pos) {
		this.start_pos = start_pos;
	}

	public long getEnd_pos() {
		return end_pos;
	}

	public void setEnd_pos(long end_pos) {
		this.end_pos = end_pos;
	}
	
	public boolean contains(Exon exon){
		if(exon.getChromosome().equals(this.chromosome)){
			if(exon.getStart_pos()>=this.start_pos && exon.getEnd_pos()<=this.end_pos){
				return true;
			}
		}
		return false;
	}

	@Override
	public String toString() {
		return "Gene [chromosome=" + chromosome + ", geneName=" + geneName
				+ ", start_pos=" + start_pos + ", end_pos=" + end_pos + "]";
	}
	
	
	
}
